In [41]:
import pandas as pd
import numpy as np

### Ej 5

In [42]:
#EJERCICIO 5

dataEj5 = [['Sem1', 20], ['Sem2', 73], ['Sem3', 75], ['Sem4', 80], ['Sem4', 82]]

dfEj5 = pd.DataFrame(dataEj5, columns=['Semana', 'CantProgs'])

print("Media", dfEj5['CantProgs'].mean())
print("Mediana", dfEj5['CantProgs'].median())
print("Moda: No hay moda porque todos los valores ocurren exactamente una vez")


Media 66.0
Mediana 75.0
Moda: No hay moda porque todos los valores ocurren exactamente una vez


### Ej 6

In [43]:
#EJERCICIO 6

dataEj6 = [['Sem1', 13], ['Sem2', 15], ['Sem3', 8], ['Sem4', 16], ['Sem4', 8]]

dfEj6 = pd.DataFrame(dataEj6, columns=['Semana', 'CantReclamos'])

print("Media:", dfEj6['CantReclamos'].mean())
print("Mediana:", dfEj6['CantReclamos'].median())
print("Moda:", dfEj6['CantReclamos'].mode()[0]) # Elegimos el 0 porque es el único valor que se repite, entonces no habrá más modas

Media: 12.0
Mediana: 13.0
Moda: 8


### Ej 7

In [44]:
Ejecutivo = [i+1 for i in range(12)]

Remuneracion = [15.8, 17.3, 28.4, 18.2, 15.0, 24.7, 13.1, 10.2, 29.3, 34.7, 16.9, 25.3]

Ejecutivo_Remuneracion = zip(Ejecutivo, Remuneracion)

dfEj7 = pd.DataFrame(Ejecutivo_Remuneracion, columns=['Ejecutivo', 'Remuneracion'])

print("Mediana:", dfEj7['Remuneracion'].median())
print("Media:", dfEj7['Remuneracion'].mean())


Mediana: 17.75
Media: 20.741666666666667


### Ej 8

In [46]:
dataEj8 = [6, 8, 7, 10, 3, 5, 9, 8]

dfEj8 = pd.DataFrame(dataEj8, columns=["Datos"])

print("Desviacion estándar:", dfEj8.std())
print("Varianza:", np.square(dfEj8.std()))

Desviacion estándar: Datos    2.267787
dtype: float64
Varianza: Datos    5.142857
dtype: float64


### Ej 9

In [69]:
dataEj9 = [10, 8, 11, 7, 9]
dfEj9 = pd.DataFrame(dataEj9, columns=["Datos"])
coeficienteVariacion = dfEj9["Datos"].std()/dfEj9["Datos"].mean()

print("Coeficiente de variacion:", coeficienteVariacion*100, "%")

Coeficiente de variacion: 17.568209223157663 %


### Ej 10

In [79]:
dataEj10 = [4.6, 3.2, 5.4, 2.6, 5,2]
wght = [8, 3, 6, 2, 5]
data_wght = [dataEj10[i]*wght[i] for i in range(5)]

data = zip(dataEj10, wght, data_wght)
dfEj10 = pd.DataFrame(data, columns=["Datos", "Peso", "DatosPonderados"])
sum_wght = dfEj10["Peso"].sum()

print("Media aritmetica sin ponderacion:", dfEj10["Datos"].mean())
print("Media aritmetica ponderada:", dfEj10["DatosPonderados"].sum()/sum_wght)

Media aritmetica sin ponderacion: 4.159999999999999
Media aritmetica ponderada: 4.541666666666667


### Ej 11

In [127]:
dataEj11 = [np.average([0, 4]), np.average([5, 9]), np.average([10, 14]), np.average([15, 19]), np.average([20, 24])]
frec = [5, 8, 11, 9, 7]

# En vez de una cadena vacía, puedes añadir un valor como promedio de los valores mencionados
dataEj11.insert(3, np.average([21, 23, 24, 21]))

data = zip(dataEj11, frec)
dfEj11 = pd.DataFrame(data, columns=["ClasePromedio","Frecuencia"])

#TODO TERMINAR!!!!

### Ejercicio 14

 La tabla adjunta muestra la cantidad de tiempo (en horas) dedicada a estudiar para un examen por una muestra aleatoria de 25 estudiantes de una clase numerosa.

a) Estime la media muestral del tiempo de estudio.

In [122]:
# Media muestral (x̄).

horasDeEstudio = [(0, 4), (4, 8), (8, 12), (12, 16), (16, 20)]
estudiantes = [3, 7, 8, 5, 2]

data = []
for intervalo, frecuencia in zip(horasDeEstudio, estudiantes):
    data.extend([(intervalo[0] + intervalo[1]) / 2] * frecuencia)

# Convierto la lista en una Series de Pandas y calculo la media muestral
serie_datos = pd.Series(data)
media_muestral = serie_datos.mean()

print(f"La media muestral del tiempo de estudio es {media_muestral:.2f} horas por estudiante.")

La media muestral del tiempo de estudio es 9.36 horas por estudiante.


b)  Estime la desviación estándar muestral. 

In [123]:
desviacion_estandar = serie_datos.std(ddof=1)  # ddof=1 para muestral

print(f"La desviación estándar muestral del tiempo de estudio es {desviacion_estandar:.3f} horas.")

La desviación estándar muestral del tiempo de estudio es 4.572 horas.


### Ejercicio 15

A continuación, se presenta una muestra aleatoria de siete pares (x, y) de
puntos de datos:
(1, 5) (3, 7) (4, 6) (3, 6) (5, 7) (5, 8)

a) Calcule la covarianza.

b) Calcule el coeficiente de correlación.

In [124]:
datos = [(1, 5), (3, 7), (4, 6), (3, 6), (5, 7), (5, 8)]

df = pd.DataFrame(datos, columns=['X', 'Y'])

covarianza = df['X'].cov(df['Y'])
print(f"Covarianza muestral: {covarianza:.3f}")

correlacion = df['X'].corr(df['Y'])
print(f"Coeficiente de correlación: {correlacion:.3f}")

Covarianza muestral: 1.300
Coeficiente de correlación: 0.817


### Ejercicio 16

Un hospital tiene interés en averiguar la eficacia de un nuevo medicamento para reducir el tiempo necesario para recuperarse totalmente de una operación de rodilla. La recuperación total se mide por medio de una serie de tests de fuerza que comparan la rodilla operada con la rodilla sin operar. El medicamento se administró en dosis diferentes a 18 pacientes durante un periodo de 6 meses. Los datos (x, y) siguientes indican el número de unidades de medicamento, X, y los días necesarios para la recuperación total Y de cada paciente:

(5, 53) (4, 56) (9, 54) (14, 56) (21, 65) (7, 53) (7, 56) (9, 59) (14, 48) (11, 66)
(21,57) (17, 49) (9, 53) (21, 52) (4, 56) (9, 46) (14, 66) (13, 49)

a) Calcule la covarianza.

b) Calcule el coeficiente de correlación.

In [125]:
datos = [(5, 53), (4, 56), (9, 54), (14, 56), (21, 65), (7, 53), (7, 56), (9, 59), (14, 48), (11, 66),
         (21, 57), (17, 49), (9, 53), (21, 52), (4, 56), 
         (9, 46), (14, 66), (13, 49)]


df = pd.DataFrame(datos, columns=['X', 'Y'])

covarianza = df['X'].cov(df['Y'])
print(f"Covarianza muestral: {covarianza:.3f}")

correlacion = df['X'].corr(df['Y'])
print(f"Coeficiente de correlación: {correlacion:.3f}")

Covarianza muestral: 4.268
Coeficiente de correlación: 0.128


c) Analice brevemente la relación entre el número de unidades de
medicamento y el tiempo de recuperación. ¿Qué dosis deberíamos
recomendar basándonos en este análisis inicial?

> El análisis de la covarianza y el coeficiente de correlación sugiere una relación muy débil entre el número de unidades de medicamento administradas y el tiempo de recuperación. La covarianza positiva de 4.031 indica una ligera tendencia a que, a medida que aumenta la dosis, aumente también el tiempo de recuperación, pero el coeficiente de correlación de 0.128 confirma que esta relación es casi nula. Esto sugiere que el número de unidades de medicamento tiene un impacto mínimo en la variación del tiempo de recuperación.
>
>Basados en estos resultados, no es posible establecer una relación clara entre la dosis del medicamento y el tiempo de recuperación. 

### Ejercicio 17

Acme Delivery ofrece tres tarifas distintas de envío de paquetes de menos de 5 libras de Maine a la costa oeste: ordinario, 3 $; urgente, 5 $, y superurgente, 10$. Para comprobar la calidad de estos servicios, un importante minorista de venta por correo envió 15 paquetes de Maine a Tacoma (Washington) en momentos elegidos aleatoriamente. Los paquetes fueron enviados en grupos de tres por los tres servicios al mismo tiempo para reducir las diferencias resultantes del día del envío. Los datos siguientes muestran el coste de envío,
X, y el número de días, Y, en pares (x, y):

(3,7) (3, 6) (3, 5) (5,5) (10,2) (3,9) (5,6) (10,5) (5, 6) (10, 1) (3, 10) (5, 7) (10, 4)
(5,6) (10, 4)

a) Describa los datos numéricamente (covarianza y correlación).

In [126]:
datos = [(3,7), (3, 6), (3, 5), (5,5), (10,2), (3,9), (5,6), (10,5), (5, 6), (10, 1), (3, 10), (5, 7), (10, 4),
(5,6),(10, 4)]
df = pd.DataFrame(datos, columns=['X', 'Y'])

covarianza = df['X'].cov(df['Y'])
print(f"Covarianza muestral: {covarianza:.3f}")

correlacion = df['X'].corr(df['Y'])
print(f"Coeficiente de correlación: {correlacion:.3f}")

Covarianza muestral: -5.500
Coeficiente de correlación: -0.776


> En este caso la covarianza nos dio negativa, lo que nos indicaría una relación inversa entre las variables X e Y, siendo que a medida que aumenta el costo del envío (X), el número de días para la entrega (Y) tiende a disminuir, lo cual es coherente. Este dato nos da una idea del comportamiento de las variables pero no el grado de variación (la proporción).
>
> Con el dato de el coeficiente de correlación de -0.776, podemos confirmar que existe una correlación negativa fuerte, dado que es muy cercana a -1. Esto significa que a medida que el costo del envío aumenta (de $3 a $10), el número de días para la entrega tiende a disminuir de manera significativa.


b) Analice el valor de los servicios de precio más alto desde el punto de vista
del envío más rápido.

>Elegir un servicio de envío premium no debe hacerse de manera apresurada. Los servicios más costosos pueden ofrecer ventajas como manejo prioritario, transporte más rápido y un seguimiento, lo que puede ser beneficioso para envíos urgentes o de alto valor, pero no siempre es indispensable pagar más para obtener una entrega rápida. Es importante considerar la urgencia del envío, su valor y las alternativas disponibles. Comparar distintas opciones permite tomar una decisión bien justificada, analizando si el costo adicional realmente justifica el beneficio de una entrega más rápida.

### Ejercicio 19

El número de computadores vendidos al día en una tienda viene definido por
la siguiente distribución de probabilidad:
\begin{array}{|c|c|c|c|c|c|c|c|}
\hline
X    & 0    & 1    & 2    & 3    & 4    & 5    & 6    \\ \hline
P(x) & 0.05 & 0.10 & 0.20 & 0.20 & 0.20 & 0.15 & 0.10 \\ \hline
\end{array}

a) ¿P(3 ≤ x ≤ 6)?:

P(3 ≤ x ≤ 6) = P(x = 3) + P(x = 4) + P(x = 5) + P(x = 6) = 0.20 + 0.20 + 0.15 + 0.10 = 0.65

b) ¿P(x > 3)?:

P(x > 3) = P(x = 4) + P(x = 5) + P(x = 6) = 0.20 + 0.15 + 0.10 = 0.45

c) ¿P(x ≤ 4)?:

P(x ≤ 4) = P(x = 0) + P(x = 1) + P(x = 2) + P(x = 3) + P(x = 4) = 0.05 + 0.10 + 0.20 + 0.20 + 0.20 = 0.75

d) ¿P(2 < x ≤ 5)?:

P(2 < x ≤ 5) = P(x = 3) + P(x = 4) + P(x = 5) = 0.20 + 0.20 + 0.15 = 0.55 
